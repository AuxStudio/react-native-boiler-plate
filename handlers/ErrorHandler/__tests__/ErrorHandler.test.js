import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import { ErrorHandler } from '..';

jest.mock('../../../scenes/pages/Error'); // Error scene that is displayed on error

describe('ErrorHandler', () => {
  let spy;
  const dispatch = jest.fn();

  function ProblemChild() {
    throw new Error('Error thrown from problem child');
    return <div>Error</div>; // eslint-disable-line
  }

  it('renders with all props', () => {
    expect(
      renderer.create(
        <ErrorHandler dispatch={jest.fn()} uid="xxxx-xxxx-xxxx-xxxx">
          <View />
        </ErrorHandler>,
      ),
    ).toMatchSnapshot();
  });

  it('renders with minimum required props', () => {
    expect(
      renderer.create(
        <ErrorHandler dispatch={jest.fn()}>
          <View />
        </ErrorHandler>,
      ),
    ).toMatchSnapshot();
  });

  it('catches errors in componentDidCatch', () => {
    spy = jest.spyOn(ErrorHandler.prototype, 'componentDidCatch');

    const component = renderer.create(
      <ErrorHandler dispatch={dispatch}>
        <ProblemChild />
      </ErrorHandler>,
    );
    const instance = component.getInstance();

    expect(instance.state.hasError).toBe(true);
    expect(component).toMatchSnapshot();
    expect(spy).toHaveBeenCalled();
    expect(dispatch).toMatchSnapshot(); // dispatch function has been called
  });

  afterEach(() => {
    if (spy) {
      spy.mockClear();
    }
  });
});
