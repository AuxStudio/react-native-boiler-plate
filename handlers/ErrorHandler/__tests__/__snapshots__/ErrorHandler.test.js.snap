// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorHandler catches errors in componentDidCatch 1`] = `null`;

exports[`ErrorHandler catches errors in componentDidCatch 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "payload": Object {
          "error": [Error: Error thrown from problem child],
          "uid": undefined,
        },
        "type": "logError",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`ErrorHandler renders with all props 1`] = `<View />`;

exports[`ErrorHandler renders with minimum required props 1`] = `<View />`;
